menuconfig.py

```diff
@@ -165,100 +165,276 @@
+
+            return int(cnum)
 
-    # Symbol information text
-    INFO_TEXT_STYLE          = LIST_STYLE
+        else:
+            raise StyleError("Unknown token: {}".format(t))
 
-    if os.environ.get("MENUCONFIG_THEME") == "aquatic":
-        # More 'make menuconfig'-like theme, contributed by Mitja Horvat
-        # (pinkfluid)
-        PATH_STYLE         = (curses.COLOR_CYAN,  curses.COLOR_BLUE,  BOLD                              )
-        SEPARATOR_STYLE    = (curses.COLOR_WHITE, curses.COLOR_CYAN,  BOLD,            curses.A_STANDOUT)
-        HELP_STYLE         = PATH_STYLE
-        DIALOG_FRAME_STYLE = SEPARATOR_STYLE
-        DIALOG_BODY_STYLE  = (curses.COLOR_WHITE, curses.COLOR_BLUE,  curses.A_NORMAL                   )
-        DIALOG_EDIT_STYLE  = (curses.COLOR_BLACK, curses.COLOR_WHITE, curses.A_NORMAL, curses.A_STANDOUT)
+    def __parse_main(tok):
+        nonlocal attrs, fg_color, bg_color
 
-    # Turn styles into attributes and store them in global variables. Doing
-    # this separately minimizes the number of curses color pairs, and shortens
-    # the style definitions a bit.
-    #
-    # Could do some locals()/globals() trickery here too, but keep it
-    # searchable.
-    _PATH_STYLE               = _style(*PATH_STYLE)
-    _SEPARATOR_STYLE          = _style(*SEPARATOR_STYLE)
-    _LIST_STYLE               = _style(*LIST_STYLE)
-    _LIST_SEL_STYLE           = _style(*LIST_SEL_STYLE)
-    _LIST_INVISIBLE_STYLE     = _style(*LIST_INVISIBLE_STYLE)
-    _LIST_INVISIBLE_SEL_STYLE = _style(*LIST_INVISIBLE_SEL_STYLE)
-    _HELP_STYLE               = _style(*HELP_STYLE)
-    _DIALOG_FRAME_STYLE       = _style(*DIALOG_FRAME_STYLE)
-    _DIALOG_BODY_STYLE        = _style(*DIALOG_BODY_STYLE)
-    _DIALOG_EDIT_STYLE        = _style(*DIALOG_EDIT_STYLE)
-    _JUMP_TO_EDIT_STYLE       = _style(*JUMP_TO_EDIT_STYLE)
-    _INFO_TEXT_STYLE          = _style(*INFO_TEXT_STYLE)
+        # Parse attributes
+        while True:
+            t = next(tok)
+
+            if t == None:
+                return
+            if t == 'bold':
+                attrs |= curses.A_NORMAL if _IS_WINDOWS else curses.A_BOLD
+            elif t == 'standout':
+                attrs |= curses.A_STANDOUT
+            elif t == 'underline':
+                attrs |= curses.A_UNDERLINE
+            else:
+                break
+
+        # Parse the color
+        fg_color = __parse_color(tok, t)
+
+        t = next(tok)
+
+        # No more tokens, exit
+        if not t:
+            return
+        # Do we have a background color?
+        elif t != "on":
+            raise StyleError("A foreground color must be followed by \"on\" to define a background color. Error near: {}".format(t))
+
+        t = next(tok)
+        bg_color = __parse_color(tok, t)
+
+        t = next(tok)
+        if t != None:
+            raise StyleError("Garbage at end of string. Error near: {}".format(t))
+
+    attrs = 0
+    fg_color = -1
```

Comment:
Instead of having `fg_color` and `bg_color` default to -1 if not specified, it would be nice if they were inherited from e.g. the theme, like for the other attributes. That way all the attributes would work the same ways.